### PR TITLE
Added 3D Scatter Plot Graph in Collections tab on click of the green "Generate 3D Graph" button.

### PR DESCRIPTION
**Pull request by RamBITS-AI**
Added 3D Scatter Plot Graph in Collections tab on click of the green "Generate 3D Graph" button.

### CODE REVIEW

File: Server/routes.go [hunk 1]
```diff
@@ -1151,18 +1153,22 @@ func (r *Routes) CreateTSNE(w http.ResponseWriter, req *http.Request) {
...
-					Logger.Log.Log("Error creating TSNE: " + err.Error())
-					return
-				}
-			}()
-			w.WriteHeader(http.StatusOK)
+			err = collection.CreateTSNE(tsne.Dimensions, tsne.Iterations, tsne.Learningrate)
+			if err != nil {
+				Logger.Log.Log("Error creating TSNE: " + err.Error())
+				return
+			} else {
```
Comment (spookieoli) [inline Server/routes.go:1160]: do you think waitingtimes are not a thing here?

**Review by spookieoli — APPROVED** [review]

File: Server/routes.go (hunk 1 above)
Comment (RamBITS-AI) [reply Server/routes.go:1160]: If we don't block the consequent call to gettsnedata will fail... How can we address this?

File: Tsne/tsne.go [hunk 2]
```diff
@@ -93,7 +93,7 @@ func (t *TSNE) computeGradients(data []*Vector.Vector) ([][]float64, error) {
 
 	for i := 0; i < n; i++ {
 		for j := 0; j < n; j++ {
-			if i != j {
+			if i != j && i < n && j < n && i < len(data) && j < len(data[i].Data) { // TODO: Check if this check (i < n && j < n && i < len(data) && j < len(data[i].Data)) is required
```
Comment (RamBITS-AI) [inline Tsne/tsne.go:96]: Please check this - @spookieoli..

Comment (spookieoli) [reply Tsne/tsne.go:96]: changed in main

Comment (RamBITS-AI) [reply Tsne/tsne.go]: Ok!

**Pull request merged**